src/helpers/http.js

```diff
@@ -0,0 +1,100 @@
+const uuid4 = require('uuid4');
+const uuidToHex = require('uuid-to-hex');
+const config = require('../config.js');
+const eventInterface = require('../event.js');
+const utils = require('../utils.js');
+
+
+const URL_BLACKLIST = {
+    'tc.epsagon.com': 'endsWith',
+    'googleapis.com': 'endsWith',
+    'amazonaws.com':
+        (url, pattern) => url.endsWith(pattern) &&
+            (url.indexOf('.execute-api.') === -1) &&
+            (url.indexOf('.es.') === -1) &&
+            (url.indexOf('.elb.') === -1) &&
+            (url.indexOf('.appsync-api.') === -1),
+    '127.0.0.1': (url, pattern, path) => (url === pattern) && path.startsWith('/2018-06-01/runtime/invocation/'),
+    '169.254.169.254': 'startsWith', // EC2 document ip. Have better filtering in the future
+};
+
+const USER_AGENTS_BLACKLIST = ['openwhisk-client-js'];
+
+/**
+ * Checks if a URL is in the user-defined blacklist.
+ * @param {string} url The URL to check
+ * @returns {boolean} True if it is in the user-defined blacklist, False otherwise.
+ */
+function isURLIgnoredByUser(url) {
+    return config.getConfig().urlPatternsToIgnore.some(pattern => url.includes(pattern));
+}
+
+
+/**
+ * Set the duration of the event, and resolves the promise using the given function.
+ * @param {object} httpEvent The current event
+ * @param {Function} resolveFunction Function that will be used to resolve the promise
+ * @param {integer} startTime The time the event started at
+ */
+function resolveHttpPromise(httpEvent, resolveFunction, startTime) {
+    httpEvent.setDuration(utils.createDurationTimestamp(startTime));
+    resolveFunction();
+}
+
+
+/**
+ * Set the duration of the event, and resolves the promise using the given function.
+ * @param {object} httpEvent The current event
+ * @param {string} key name in metadata
+ * @param {string} data data to jsonify
+ */
+function setJsonPayload(httpEvent, key, data) {
+    try {
+        const payload = JSON.parse(data);
+        eventInterface.addToMetadata(httpEvent, {}, {
+            [key]: payload,
+        });
```

Comment:
no need to `JSON.parse` the data (actually, you shouldn't even since the proto object get a str-str map)

Reply:
1. I think we should capture only JSON data and not free text
2. It will be parsed more easily in Epsagon